tests/Http/Offer/SelectMainImageRequestHandlerTest.php

```diff
@@ -152,6 +149,35 @@ public function it_throws_when_image_does_not_exist(): void
         );
     }
 
+    /**
+     * @test
+     */
+    public function it_throws_when_image_is_not_linked_to_offer(): void
+    {
+        $this->givenThereIsAnImage();
+
+        $this->commandBus->throwsOnDispatch(new ImageMustBeLinkedException());
```

Comment:
Nit: As an alternative you could just create another instance of `SelectMainImageRequestHandler` in this test method instead of using `$this->selectMainImageRequestHandler`, and inject a mock `CommandBus` that throws an exception when `dispatch()` is called. Then you don't need to create a new `ThrowingTraceableCommandBus` class specifically for this one test case.